diff --git a/os_dep/osdep_service.c b/os_dep/osdep_service.c
index 7367bca..dcc07a8 100644
--- a/os_dep/osdep_service.c
+++ b/os_dep/osdep_service.c
@@ -2575,7 +2575,7 @@ inline u32 rtw_random32(void)
 {
 #ifdef PLATFORM_LINUX
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0))
-	return prandom_u32();
+	return get_random_u32();
 #elif (LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 18))
 	u32 random_int;
 	get_random_bytes(&random_int , 4);
